cmd/pulumi-test-language/tests.go

```diff
@@ -984,6 +984,95 @@ var languageTests = map[string]languageTest{
 			},
 		},
 	},
+	// This test checks how SDKs propagate properties marked as secret to the provider Configure on the gRPC level.
+	"l2-provider-grpc-config-schema-secret": {
+		providers: []plugin.Provider{&providers.ConfigGrpcProvider{}},
+		runs: []testRun{
+			{
+				assert: func(l *L,
+					projectDirectory string, err error,
+					snap *deploy.Snapshot, changes display.ResourceChanges,
+				) {
+					g := &grpcTestContext{l: l, s: snap}
+
+					// Verify the CheckConfig request received by the provider.
+					r := g.CheckConfigReq("config")
+
+					// CheckConfig request gets the secrets in the plain. This is suspect, probably
+					// has to do with secret negotiation happening later in the gRPC provider cycle.
```

Comment:
Does this need to be fixed, and should we open an issue for it?

Reply:
I think that might be worth it for consistency, though providers have adapted somehow to the current behavior, so I do not know how pressing this is. https://github.com/pulumi/pulumi/issues/16876 may be tracking handshake changes that would be necessary. Currently I suspect the engine does not send secrets to CheckConfig, because it is the result of Configure that's post CheckConfig where the provider advertises the supportsSecrets capability.